File: lib/commands/tomorrow-command.js
'use strict';

const BaseCommand = require('./base-command');
const moment = require('moment');
const Menu = require('../../data/models/menu');

class TomorrowCommand extends BaseCommand {
  async handle() {
    const menu = await Menu.findOne({ date: moment().add(1, 'days').format('YYYYMMDD') });
    if (!menu)
      return 'I don\'t know which meals are being served tomorrow!';
    let message = `Tomorrow is the *${moment().add(1, 'days').format('DD.MM.YYYY')}* and the meals are:\n`;
    menu.meals.forEach(meal => {
      message += `  ${meal}\n`;
    });
    return message;
  }
}

module.exports = TomorrowCommand;

File: src/Deathstar.Canteen/Commands/TomorrowCommand.cs
using System;
using System.Threading;
using System.Threading.Tasks;
using Deathstar.Canteen.Commands.Abstractions;
using Deathstar.Canteen.Persistence;
using Deathstar.Canteen.Slack;

namespace Deathstar.Canteen.Commands
{
	public class TomorrowCommand : ICommand
	{
		private readonly IMenuCollection menuCollection;
		private readonly ISlackbot slackbot;

		public TomorrowCommand(IMenuCollection menuCollection, ISlackbot slackbot)
		{
			this.menuCollection = menuCollection;
			this.slackbot = slackbot;
		}

		public string HelpText { get; } = "The *tomorrow* command will return a list of tomorrow's meals.";

		public string Name { get; } = "tomorrow";

		public async Task HandleAsync(ICommandMessage message, CancellationToken cancellationToken)
		{
			DateTime date = DateTime.Today.AddDays(1);
			Menu menu = await menuCollection.SingleOrDefaultAsync(x => x.Date == date.ToString("yyyyMMdd"), cancellationToken);
			string response = menu == null
				? "I don't know which meals are being served tomorrow!"
				: $"Tomorrow is the *{date:dd.MM.yyyy}* and the meals are:{Environment.NewLine}{menu}";

			slackbot.SendMessage(message.Channel, response);
		}
	}
}
